dockerRunner.go

```diff
@@ -1087,3 +858,232 @@ func (dr *dockerRunnerImpl) DeleteBridgeNetwork(ctx context.Context) error {
+	if err != nil {
+		return "", extension, err
+	}
+
+	// create target file to render template to
+	targetFile, err := ioutil.TempFile(dr.entrypointTargetDir, "estafette-entrypoint-*.sh")
+	if err != nil {
+		return "", extension, err
+	}
+	defer targetFile.Close()
+	path = targetFile.Name()
+
+	err = entrypointTemplate.Execute(targetFile, data)
+	if err != nil {
+		return "", extension, err
+	}
+
+	if runtime.GOOS != "windows" {
+		err = os.Chmod(path, 0755)
+		if err != nil {
+			return "", extension, err
+		}
+	}
+
+	bytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", extension, err
+	}
+	log.Debug().Str("entrypoint", string(bytes)).Msgf("Inspecting entrypoint script at %v", path)
+
+	return path, extension, nil
+}
+
+func (dr *dockerRunnerImpl) initContainerStartVariables(shell string, commands []string, commandsNotAsJob bool) (entrypoint []string, cmds []string, binds []string) {
+	entrypoint = make([]string, 0)
+	cmds = make([]string, 0)
+	binds = make([]string, 0)
+
+	if len(commands) > 0 {
+		// generate entrypoint script
+		path, extension, err := dr.generateEntrypointScript(shell, commands, commandsNotAsJob)
+		if runtime.GOOS != "windows" && err == nil {
+			// use generated entrypoint script for executing commands
+			entrypointScriptPath := fmt.Sprintf("/estafette-entrypoint.%v", extension)
+			entrypoint = []string{entrypointScriptPath}
+			binds = append(binds, fmt.Sprintf("%v:%v", path, entrypointScriptPath))
+		} else {
+			// generating entrypoint script failed, do it in the old way
+			log.Warn().Err(err).Msgf("Generating entrypoint script failed, setting up entrypoint and command in old style")
+
+			// define entrypoint
+			entrypoint = []string{shell}
+			if runtime.GOOS == "windows" && shell == "powershell" {
+				entrypoint = append(entrypoint, "-Command")
+			} else if runtime.GOOS == "windows" && shell == "cmd" {
+				entrypoint = append(entrypoint, "/S", "/C")
+			} else {
+				entrypoint = append(entrypoint, "-c")
+			}
+
+			// define commands
+			cmdStopOnErrorFlag := ""
+			cmdSeparator := ";"
+			if runtime.GOOS == "windows" && shell == "powershell" {
+				cmdStopOnErrorFlag = "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; "
+				if dr.config.DockerDaemonMTU != nil && *dr.config.DockerDaemonMTU != "" {
+					mtu, err := strconv.Atoi(*dr.config.DockerDaemonMTU)
+					if err == nil {
+						mtu -= 50
+						cmdStopOnErrorFlag += fmt.Sprintf("Write-Host 'Updating MTU to %v...'; Get-NetAdapter | Where-Object Name -like \"*Ethernet*\" | ForEach-Object { & netsh interface ipv4 set subinterface $_.InterfaceIndex mtu=%v store=persistent }; ", mtu, mtu)
+					}
+				}
+				cmdSeparator = ";"
+			} else if runtime.GOOS == "windows" && shell == "cmd" {
+				cmdStopOnErrorFlag = ""
+				cmdSeparator = " && "
+			} else {
+				cmdStopOnErrorFlag = "set -e; "
+				cmdSeparator = ";"
+			}
+			cmds = append(cmds, cmdStopOnErrorFlag+strings.Join(commands, cmdSeparator))
```

Comment:
so even though docker allows to have multiple commands we are putting them all in one line "command".
Is it like this to run all in the same shell? (I'm not really familiar how docker deals with this, hence my question)

Reply:
This is the old style way of running the commands. That has worked for a long time, it just shows up in this PR because it's now nested in an if.

Reply:
By default docker's entrypoint is `["/bin/sh","-c"]` or something along those lines; multiple commands are passed as arguments as far as I know.